### PR TITLE
DAOController fixes: C1, C2, C6, C9 & C8

### PR DESCRIPTION
**Pull request by MiltonTulli**
### Description
DAOController: 
- C1. Allow specifying the `paramshash` that is passed to the initialize function.
- C2. Fix schemesWithManageSchemesPermission counter on `registerScheme` fn 
- C6. Add batching requests for getActiveProposals & getInactiveProposals. 
- C8. Prevent scheme to end proposal of another scheme
- C9. Verify that a proposal ID is not already used by another scheme at startProposal()
- Add DaoController tests for new features/fixes
- Add getters for active and inactive proposals count.

Closes https://github.com/DXgovernance/dxdao-contracts/issues/226
Closes https://github.com/DXgovernance/dxdao-contracts/issues/227
Closes https://github.com/DXgovernance/dxdao-contracts/issues/232
Closes https://github.com/DXgovernance/dxdao-contracts/issues/230
Closes https://github.com/DXgovernance/dxdao-contracts/issues/233

Screen Shot 2022-11-04 at 21 21 14

### CODE REVIEW

File: contracts/dao/DAOController.sol [hunk 1]
```diff
@@ -211,25 +226,81 @@ contract DAOController is Initializable {
...
+        view
+        returns (ProposalAndScheme[] memory activeProposalsArray)
+    {
+        return _getProposalsBatchRequest(_start, _end, activeProposals);
+    }
+
+    /**
+     * @dev Returns array of inactive proposals
+     * @param _start index to start batching
+     * @param _end last index of batch
```
Comment (AugustoL) [inline contracts/dao/DAOController.sol:279]: * @param _end last index of batch (included). Zero will return all

File: contracts/dao/DAOController.sol [hunk 2]
```diff
@@ -211,25 +226,81 @@ contract DAOController is Initializable {
...
+        external
+        view
+        returns (ProposalAndScheme[] memory activeProposalsArray)
+    {
+        return _getProposalsBatchRequest(_start, _end, activeProposals);
+    }
+
+    /**
+     * @dev Returns array of inactive proposals
+     * @param _start index to start batching
```
Comment (AugustoL) [inline contracts/dao/DAOController.sol:278]: * @param _start index to start batching (included).

**Review by AugustoL — CHANGES_REQUESTED** [review]

File: contracts/dao/DAOController.sol (hunk 1 above)
Comment (MiltonTulli) [reply contracts/dao/DAOController.sol]: Done!

File: contracts/dao/DAOController.sol (hunk 2 above)
Comment (MiltonTulli) [reply contracts/dao/DAOController.sol]: Done!

**Review by AugustoL — APPROVED** [review]

**Pull request merged**